sqld/src/database/dump/exporter.rs

```diff
@@ -420,10 +415,11 @@ fn find_unused_str(haystack: &str, needle1: &str, needle2: &str) -> String {
     }
 }
 
-pub fn export_dump(mut db: rusqlite::Connection, writer: impl Write) -> anyhow::Result<()> {
-    let mut txn = db.transaction()?;
-    txn.execute("PRAGMA writable_schema=ON", ())?;
-    let savepoint = txn.savepoint_with_name("dump")?;
+pub fn export_dump(mut db: libsql::Connection, writer: impl Write) -> anyhow::Result<()> {
+    db.execute("BEGIN", ())?;
+    db.execute("PRAGMA writable_schema=ON", ())?;
+    // FIXME: savepoint logic is lost during the out-of-rusqlite migration, we need to restore it
```

Comment:
Is this something we should add to the libsql issue tracking missing features for the rust api?

Reply:
Yes, definitely. We generally miss any kind of Transaction API, that lets you automatically commit/rollback, have a well defined shutdown path, and so on. Basically https://docs.rs/rusqlite/latest/rusqlite/struct.Transaction.html